invokeai/app/services/invocation_processor/invocation_processor_default.py

```diff
@@ -195,6 +194,7 @@ def __process(self, stop_event: Event):
                             error=traceback.format_exc(),
                         )
                 elif is_complete:
+                    self.__invoker.services.performance_statistics.log_stats(graph_execution_state.id)
```

Comment:
I'm not super familiar with this code. @psychedelicious is there a better place for this?

Reply:
This is earliest we know a graph is complete, I think it's a reasonable spot. The only other area would be in `invokeai/app/services/session_queue/session_queue_sqlite.py` in the handlers for complete/failed/canceled session queue items.

Unrelated, but are we resetting the stats correctly when a graph is canceled?

Reply:
We try to reset when we know that we should, but I don't have high confidence that we reset in all cancellation and error cases. This is why I've kept `_prune_stale_stats(...)` for now. Maybe its just my lack of familiarity with `invocation_processor_default.py`, but I had a hard time tracing all of the possible termination paths.

Reply:
It's not lack of familiarity - it's a bit convoluted. Needs a bit of TLC.